app/components/canvas/ArticleViewModal.tsx

```
@@ -0,0 +1,228 @@
+
+          {/* Article Content - Scrollable Area */}
+          <div className="flex-1 overflow-hidden">
+            <ScrollArea className="h-full">
+              <div className="p-6">
+                {article ? (
+                  isMarkdown ? (
+                    searchQuery ? (
+                      // For search in markdown, show plain text with highlights
+                      <div 
+                        className="prose prose-sm dark:prose-invert max-w-none whitespace-pre-wrap leading-relaxed"
+                        dangerouslySetInnerHTML={{ __html: highlightedText }}
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)